Sources/Confidence/EventSenderEngine.swift

```diff
@@ -22,6 +22,7 @@ final class EventSenderEngineImpl: EventSenderEngine {
     private let uploader: ConfidenceClient
     private let clientSecret: String
     private let payloadMerger: PayloadMerger = PayloadMergerImpl()
+    private let semaphore = DispatchSemaphore(value: 1)
```

Comment:
I haven't verified this, but would be possible to configure the upload channel to be serial instead of adding an extra semaphore inside?

```
let serialUploadReqChannel = uploadReqChannel.receive(on: serialQueue)
```

Reply:
this should work too and I didn't know about this. nice one! if you could verify that this works, i'm fine replacing the semaphore.